### ISSUE TITLE
[🐛 Bug] 4.11.2: pytest is failing in `test/unit/selenium/webdriver/ie/test_ie_options.py::test_element_scroll_behavior` unit

### ISSUE THREAD

**Issue by kloczek** [SeleniumHQ/selenium#12643]
### What happened?

Looks like something is wrong in `test/unit/selenium/webdriver/ie/test_ie_options.py::test_element_scroll_behavior` unit.

### How can we reproduce the issue?

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix> using 'installer` module
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

Here is list of installed modules in build env
<details>

```
Package                       Version
----------------------------- ---------
alabaster                     0.7.13
async-generator               1.10
attrs                         23.1.0
Babel                         2.12.1
build                         0.10.0
certifi                       2023.7.22
charset-normalizer            3.2.0
distro                        1.8.0
docutils                      0.20.1
exceptiongroup                1.1.3
gpg                           1.21.0
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.8.0
iniconfig                     2.0.0
installer                     0.7.0
Jinja2                        3.1.2
libcomps                      0.1.19
MarkupSafe                    2.1.3
outcome                       1.2.0
packaging                     23.1
pkg                           0.0.0
pluggy                        1.3.0
Pygments                      2.16.0
pyproject_hooks               1.0.0
pytest                        7.4.0
pytest-mock                   3.11.1
pytest-trio                   0.8.0
python-dateutil               2.8.2
pytz                          2023.2
requests                      2.31.0
setuptools                    68.0.0
six                           1.16.0
sniffio                       1.3.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
Sphinx                        7.0.1
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.3
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.9
tomli                         2.0.1
trio                          0.22.0
urllib3                       2.0.4
wheel                         0.41.1
zipp                          3.16.2
```
</details>

### Relevant log output

<details>

```
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-selenium-4.11.2-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-selenium-4.11.2-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network' -q

test/unit/selenium/webdriver/ie/test_ie_options.py::test_arguments PASSED [  3%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_browser_attach_timeout PASSED [  6%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_raises_exception_for_invalid_browser_attach_timeout PASSED [  9%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_element_scroll_behavior FAILED [ 12%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_ensure_clean_session PASSED [ 16%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_file_upload_dialog_timeout PASSED [ 19%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_raises_exception_for_file_upload_dialog_timeout PASSED [ 22%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_force_create_process_api PASSED [ 25%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_force_shell_windows_api PASSED [ 29%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_full_page_screenshot PASSED [ 32%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_ignore_protected_mode_settings PASSED [ 35%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_ignore_zoom_level PASSED [ 38%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_initial_browser_url PASSED [ 41%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_native_events PASSED [ 45%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_persistent_hover PASSED [ 48%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_require_window_focus PASSED [ 51%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_use_per_process_proxy PASSED [ 54%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_use_legacy_file_upload_dialog_handling PASSED [ 58%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_attach_to_edge_chrome PASSED [ 61%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_edge_executable_path PASSED [ 64%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_additional_options PASSED [ 67%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_to_capabilities PASSED [ 70%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_to_capabilities_arguments PASSED [ 74%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_to_capabilities_additional_options PASSED [ 77%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_to_capabilities_should_not_modify_set_options PASSED [ 80%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_starts_with_default_capabilities PASSED [ 83%]
test/unit/selenium/webdriver/ie/test_ie_options.py::test_is_a_baseoptions PASSED [ 87%]
test/unit/selenium/webdriver/virtual_authenticator/credentials_test.py::test_rk_enabled_credential PASSED [ 90%]
test/unit/selenium/webdriver/virtual_authenticator/credentials_test.py::test_rk_disabled_credentials PASSED [ 93%]
test/unit/selenium/webdriver/virtual_authenticator/credentials_test.py::test_to_dict PASSED [ 96%]
test/unit/selenium/webdriver/virtual_authenticator/credentials_test.py::test_from_dict PASSED [100%]

=================================== FAILURES ===================================
_________________________ test_element_scroll_behavior _________________________

opts = <selenium.webdriver.ie.options.Options object at 0x7f383fafbeb0>

    def test_element_scroll_behavior(opts):
>       opts.element_scroll_behavior = ElementScrollBehavior.BOTTOM

test/unit/selenium/webdriver/ie/test_ie_options.py:53:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.ie.options._IeOptionsDescriptor object at 0x7f3840f9dbb0>
obj = <selenium.webdriver.ie.options.Options object at 0x7f383fafbeb0>
value = <ElementScrollBehavior.BOTTOM: 1>

    def __set__(self, obj, value) -> None:
        if not isinstance(value, self.expected_type):
>           raise ValueError(f"{self.name} should be of type {self.expected_type.__name__}")
E           ValueError: elementScrollBehavior should be of type int

selenium/webdriver/ie/options.py:59: ValueError
=========================== short test summary info ============================
FAILED test/unit/selenium/webdriver/ie/test_ie_options.py::test_element_scroll_behavior
========================= 1 failed, 30 passed in 0.36s =========================
```

### Operating System

Linux x86/64

### Selenium version

4.11.2

### What are the browser(s) and version(s) where you see this issue?

N/A

### What are the browser driver(s) and version(s) where you see this issue?

N/A

### Are you using Selenium Grid?

N/A

**Comment by titusfortner** [comment]
It's a type error that was fixed in 4.12. Thanks for reporting it, next release out soon.

**Comment by kloczek** [comment]
No problems .. thing happens sometimes 😋 
Thank you 👍 😄
